src/libawkward/array/EmptyArray.cpp

```diff
@@ -125,7 +125,7 @@ namespace awkward {
   }
 
   const std::shared_ptr<Content> EmptyArray::flatten(int64_t axis) const {
-    throw std::runtime_error("FIXME: not implemented");
+    throw std::invalid_argument("array contains no Records");
```

Comment:
I think that flattening an empty array should return an empty array. I just tried it in Scala:

```scala
scala> List[List[Int]]()
res12: List[List[Int]] = List()

scala> List[List[Int]]().flatten
res13: List[Int] = List()
```

Also, NumPy can `reshape` empty arrays (its closest equivalent of `flatten`), so I think this should just return

```c++
std::make_shared<EmptyArray>(Identities::none(), util::Parameters())
```

We'll need to address `Identities` everywhere (later, before or after addressing `axis`), but any `Parameters` this array has should be lost because flattening does not return the same type (`List[List[Int]] → List[Int]` in the Scala example).

Reply:
```c++
   const std::shared_ptr<Content> EmptyArray::flatten(int64_t axis) const {
-    throw std::invalid_argument("array contains no Records");
+    return shallow_copy();
   }

```

Reply:
I didn't suggest `shallow_copy()` because the type should change, and that's reflected in any `Parameters` the array might have. We want the output `Parameters` to be empty, and in principle, there's a minor change that should go into the `Identities` as well. Those are both "second-order" (or even "third-order") in importance, but we might as well have a foothold into a return value that is not a strict copy.